bpf/flows.c

```diff
@@ -171,7 +180,53 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
     }
     id.if_index = skb->ifindex;
     id.direction = direction;
+    u16 flags_t = 0;
+    if ((void *)eth + sizeof(struct ethhdr) <= data_end) {
+        if (bpf_ntohs(eth->h_proto) == ETH_P_IP) {
+            const struct iphdr *iph = (struct iphdr *)(data + sizeof(struct ethhdr));
+            if ((void *)iph + sizeof(struct iphdr) <= data_end) {
+                if (iph->protocol == IPPROTO_TCP) {
+                    const struct tcphdr *th = (struct tcphdr *)(data + sizeof(struct iphdr));
+                    if ((void *)th + sizeof(struct tcphdr) <= data_end) {
+                        bpf_printk("fin=%d, syn=%d, rst=%d, psh=%d, ack=%d, urg=%d, ece=%d, cwr=%d, res1=%d doff=%d\n",th->fin, th->syn, th->rst, th->psh, th->ack,th->urg,th->ece, th->cwr, th->res1, th->doff);
```

Comment:
Are we sure we want to log a message for each packet? Apart of the overhead I'm afraid the logs will get flooded.